src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -0,0 +1,925 @@
+                         trajectory.trajectory_elements[i + 1].position.x,
+                     2) +
+                 pow(trajectory.trajectory_elements[i].position.y -
+                         trajectory.trajectory_elements[i + 1].position.y,
+                     2)));
+        segment_length_sum += length;
+    }
+    EXPECT_NEAR(segment_length_sum, arc_segment_length, 0.01);
+}
+
+TEST_F(TrajectoryPlannerTest, check_end_points_match_path)
+{
+    Polynomial2dOrder3_t path = {
+        .x = {.coefficients = {2, 0, 1, 0}},
+        .y = {.coefficients = {1, 0, 1, 0}},
+
+    };
+
+    FirmwareRobotPathParameters_t path_parameters;
+    path_parameters.path    = path;
+    path_parameters.t_start = 0;
+    path_parameters.t_end   = 1;
+    path_parameters.num_segments =
+        TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS;  // For tests the number of segments is
+                                              // arbitrary, so use the maximum possible
+    path_parameters.max_allowable_acceleration = 3;
+    path_parameters.max_allowable_speed        = 3;
+    path_parameters.initial_speed              = 0;
+    path_parameters.final_speed                = 0;
+
+    TrajectoryElement_t const_arc_elements[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    Trajectory_t trajectory = {.trajectory_elements = const_arc_elements};
+
+    TrajectoryPlannerGenerationStatus status =
+        app_trajectory_planner_generateConstantArcLengthTrajectory(path_parameters,
+                                                                   &trajectory);
+    EXPECT_EQ(OK, status);
+
+    // Create the parmeterization to contain the desired number of segments
+    CREATE_STATIC_ARC_LENGTH_PARAMETRIZATION(arc_length_param,
+                                             TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS);
+    // Get all of the points for the arc length parameterization (Not constant arc length
+    // segments)
+    shared_polynomial_getArcLengthParametrizationOrder3(
+        path_parameters.path, path_parameters.t_start, path_parameters.t_end,
+        arc_length_param);
+
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[path_parameters.num_segments - 1].position.x,
+        shared_polynomial2d_getValueOrder3(path_parameters.path, path_parameters.t_end).x,
+        0.01);
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[path_parameters.num_segments - 1].position.y,
+        shared_polynomial2d_getValueOrder3(path_parameters.path, path_parameters.t_end).y,
+        0.01);
+
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[0].position.x,
+        shared_polynomial2d_getValueOrder3(path_parameters.path, path_parameters.t_start)
+            .x,
+        0.01);
+    EXPECT_NEAR(
+        trajectory.trajectory_elements[0].position.x,
+        shared_polynomial2d_getValueOrder3(path_parameters.path, path_parameters.t_start)
+            .y,
+        0.01);
+}
+
+TEST_F(TrajectoryPlannerTest, dynamics_dont_exceed_maximums_straight_line_acceleration3)
```

Comment:
same comment about naming here. Please rename to get rid of the "3" at the end.

Reply:
Changed to 'medium acceleration'